expression/builtin_miscellaneous.go

```diff
@@ -1043,5 +1044,46 @@ type uuidShortFunctionClass struct {
 }
 
 func (c *uuidShortFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "UUID_SHORT")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETInt)
+	if err != nil {
+		return nil, err
+	}
+	bf.tp.Charset, bf.tp.Collate = ctx.GetSessionVars().GetCharsetInfo()
+	bf.tp.Flen = 36
+	sig := &builtinUUIDShortSig{bf}
+	return sig, nil
+}
+
+type builtinUUIDShortSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinUUIDShortSig) Clone() builtinFunc {
+	newSig := &builtinUUIDShortSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+var increment int64
+
+// evalInt evals a builtinUUIDShortSig.
+// See https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-short
+func (b *builtinUUIDShortSig) EvalInt(_ chunk.Row) (d int64, isNull bool, err error) {
+	
+	var uuidShort int64
+	time, _, err := uuid.GetTime()
+	if err != nil {
+		return uuidShort, false, err
+	}
+
+	buf := make([]byte, 2, 8)
+	buf = append(buf, uuid.NodeID()...)
+	var serverID = binary.BigEndian.Uint64(buf)
```

Comment:
I found  the way to get `server_id` yesterday, I think we can use followed
```
// evalInt evals a builtinUUIDShortSig.
// See https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-short
func (b *builtinUUIDShortSig) evalInt(_ chunk.Row) (d int64, isNull bool, err error) {
	now, seq, err := uuid.GetTime()
	if err != nil {
		return d, false, err
	}
	serverID := b.ctx.GetSessionVars().ConnectionInfo.ServerID
	d = int64(serverID&255)<<56 + int64(now<<16) + int64(seq)
	return d, false, nil
}
```
because of the `seq` is only a `int16` variable, we can use `now<<16` there.